metpy/calc/thermo.py

```diff
@@ -274,6 +314,199 @@ def parcel_profile(pressure, temperature, dewpt):
 
 
 @exporter.export
+def virtual_temperature(pressure, temperature, dewpt):
+    r"""Calculate virtual temperature.
+
+    Description
+
+    Parameters
+    ----------
+    pressure : `pint.Quantity`
+        The atmospheric pressure level(s) of interest. The first entry should be the starting
+        point pressure.
+    temperature : `pint.Quantity`
+        The starting temperature
+    dewpt : `pint.Quantity`
+        The starting dew point
+
+    Returns
+    -------
+    `pint.Quantity`
+        The virtual temperature.
+
+    See Also
+    --------
+    mixing_ratio, saturation_vapor_pressure
+    """
+    Rv = mixing_ratio(saturation_vapor_pressure(dewpt), pressure)
+    return temperature * (1 + Rv / epsilon) / (1 + Rv)
+
+def _find_append_zero_crossings(x, y):
+    r"""
+    Find and interpolate zero crossings
+
+    Estimate the zero crossings of an x,y series and add estimated crossings to series,
+    returning a sorted array with no duplicate values.
+
```

Comment:
Over and above what's needed in general for internal helpers--but certainly keep this since you're written it!